docs/tutorial/snapcraft-guide.md

```diff
@@ -0,0 +1,181 @@
+├── locales
+│   ├── am.pak
+│   ├── ar.pak
+│   ├── [...]
+├── natives_blob.bin
+├── node.dll
+├── resources
+│   ├── app
+│   └── atom.asar
+├── snapshot_blob.bin
+├── squirrel.exe
+└── ui_resources_200_percent.pak
+```
+
+## Step 2: Running electron-installer-snap
+
+From a terminal that has `snapcraft` in its `PATH`, run `electron-installer-snap`
+with the only required parameter `--out`, which is the location of your packaged
+Electron application created in the first step.
+
+```sh
+npx electron-installer-snap --src=out/myappname-linux-x64
+```
+
+If you have an existing build pipeline, you can use `electron-installer-snap`
+programmatically. For more information, see the API docs.
+
+```js
+const snap = require('electron-installer-snap')
+
+snap(options)
+  .then(snapPath => console.log(`Created snap at ${snapPath}!`))
+```
+
+# Using an Existing Debian Package
+
+Snapcraft is capable of taking an existing `.deb` file and turning it into
+a `.snap` file. The creation of a snap is configured using a `snapcraft.yaml`
+file that describes the sources, dependencies, description, and other core
+building blocks.
+
+## Step 1: Create a Debian Package
+
+If you do not already have a `.deb` package, using `electron-installer-snap`
+might be an easier path to create snap packages. However, multiple solutions
+for creating Debian packages exist, including [`electron-forge`](electron-forge),
+[`electron-builder`]() or [`electron-installer-debian`](electron-installer-debian).
+
+## Step 2: Create a snapcraft.yaml
+
+For more information on the available configuration options, see the
+[documentation on the snapcraft syntax](https://docs.snapcraft.io/build-snaps/syntax).
+In this example
+
+```yaml
+name: myApp
+version: 2.0.0
+summary: A little description for the app.
+description: |
+ You know what? This app is amazing! It does all the things
+ for you. Some say it keeps you young, maybe even happy.
+
+grade: stable
+confinement: classic
+
+parts:
+  slack:
+    plugin: dump
+    source: my-deb.deb
+    source-type: deb
+    after:
+      - desktop-gtk2
+    build-packages:
+      - patchelf
+    stage-packages:
+      - libasound2
+      - libgconf2-4
+      - libnotify4
+      - libnspr4
+      - libnss3
+      - libpcre3
+      - libpulse0
+      - libxss1
+      - libxtst6
+  electron-launch:
+    plugin: dump
+    source: files/
+    prepare: |
+      chmod +x bin/electron-launch
+
+apps:
+  myApp:
+    command: bin/electron-launch $SNAP/usr/lib/myApp/myApp
+    desktop: usr/share/applications/myApp.desktop
+    # Correct the TMPDIR path for Chromium Framework/Electron to ensure
+    # libappindicator has readable resources.
+    environment:
+      TMPDIR: $XDG_RUNTIME_DIR
+```
+
+As you can see, the `snapcraft.yaml` instructs the system to launch a file
+called `electron-launch`. In this example, it simply passes information on
+to the app's binary:
+
+```sh
+#!/bin/sh
+
+exec "$@" --executed-from="$(pwd)" --pid=$$ > /dev/null 2>&1 &
+```
```

Comment:
My understanding is that `bin/electron-launch` is only necessary for classic confinement, otherwise it should use `desktop-launch`.